### ISSUE TITLE
[Bug]: Substack RSS Feed Returns 403 Error During GitHub Actions Workflow

### ISSUE THREAD

**Issue by veyselserifoglu** [gautamkrishnar/blog-post-workflow#267]
### Describe the bug
When trying to fetch a Substack RSS feed using GitHub Actions, the request returns a 403 Forbidden error. This occurs despite setting custom User-Agent and Accept headers that work successfully in local environments.

### Expected behavior
The RSS feed should be fetched successfully by the GitHub Actions workflow, just as it does when the same curl command is run locally. The expectation is that the custom headers and User-Agent would bypass any bot protection.

### Screenshots

Here is a snippet of the GitHub workflow action failing: 

![image](https://github.com/user-attachments/assets/9ed379ea-bf0c-44db-9a35-66a9a774af15)

It is worth mentioning that fetching the substack rss feed locally by running curl -I substack.com/feed, returns the 403 error code as well. 

![image](https://github.com/user-attachments/assets/0fa18a32-c03f-49f5-9c60-1a61ccf0f33c)

### Workflow Yml Used
```yml

name: Latest blog post workflow
on:
  schedule: # Run workflow automatically
    - cron: '0 * * * *' # Runs every hour, on the hour
  workflow_dispatch:
permissions:
  contents: write # To write the generated contents to the readme

jobs:
  update-readme-with-blog:
    name: Update this repo's README with latest blog posts
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v4

      - name: Pull in substack posts
        uses: gautamkrishnar/blog-post-workflow@v1
        with:
          feed_list: "https://vserifoglu.substack.com/feed"
          user_agent: "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:129.0) Gecko/20100101 Firefox/129.0"
          accept_header: "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/png,image/svg+xml,*/*;q=0.8"

      - name: Start tmate session for debugging
        if: failure() # Only start tmate if the previous step fails
        uses: mxschmitt/action-tmate@v3
```

### Additional context
I opened a tmate session and connected to the GitHub Actions runner via SSH as described [here](https://github.com/marketplace/actions/debugging-with-tmate). From there, I manually executed the curl command to fetch the Substack RSS feed, and it worked without any issues. This suggests that the problem might not be related to the service blocking GitHub’s IP addresses outright. However, the 403 Forbidden error persists when the same request is made during the normal execution of the GitHub Actions workflow.

### Double check the following
Before opening a bug, Please make sure that you had done the following, add a checkmark if done by marking it with `[x]`:

- [x] I had gone through the [FAQs](https://github.com/gautamkrishnar/blog-post-workflow#faqs-and-common-issues) 
- [x] I had followed all the steps mentioned in the [documentation](https://github.com/gautamkrishnar/blog-post-workflow)

**Comment by veyselserifoglu** [comment]
After I submitted the issue, I realized that I should have searched through the closed issues first. I discovered that a similar issue had already been opened and discussed in [Issue #240](https://github.com/gautamkrishnar/blog-post-workflow/issues/240). It appears that the problem is related to how Substack handles requests, and it’s not something that can be resolved directly within the GitHub Actions workflow.
